lib/tables.c

```diff
@@ -4675,3 +4675,146 @@ table_collection_simplify(table_collection_t *self,
     simplifier_free(&simplifier);
     return ret;
 }
+
+
+/*******************
+ * Cleaning tables *
+ ******************/
+
+typedef struct {
+    site_table_t *sites;
+    mutation_table_t *mutations;
+    /* Map of old site IDs to new site IDs. */
+    site_id_t *site_id_map;
+} table_cleaner_t;
+
+static int table_cleaner_run(table_cleaner_t *self)
+{
+    // Remove any sites with duplicate positions, retaining only the *first*
+    // one. Assumes the tables have been sorted, throwing an error if not.
+    // TODO: this could be done in place rather than by constructing a new
+    // table.
+
+    int ret = 0;
+
+    if (self->sites->num_rows <= 1) {
+        // nothing to do
+        goto out;
```

Comment:
Slightly uncomfortable with this --- gotos should really only be used for error handling, and this feels like control flow. Do we really need to make this special case here? Usually if the logic is correct the empty-table case will just work without problems.